pkg/gardenlet/controller/shoot/shoot/reconciler_reconcile.go

```diff
@@ -446,10 +447,20 @@ func (r *Reconciler) runReconcileShootFlow(ctx context.Context, o *operation.Ope
 				DoIf(gardencorev1beta1helper.GetShootServiceAccountKeyRotationPhase(o.Shoot.GetInfo().Status.Credentials) == gardencorev1beta1.RotationCompleting),
 			Dependencies: flow.NewTaskIDs(initializeShootClients, waitUntilKubeControllerManagerReady),
 		})
+		deleteBastions = g.Add(flow.Task{
+			Name:         "Deleting Bastions",
+			Fn:           flow.TaskFn(botanist.DeleteBastions).SkipIf(shootSSHAccessEnabled),
+			Dependencies: flow.NewTaskIDs(deployReferencedResources, waitUntilInfrastructureReady, waitUntilControlPlaneReady),
+		})
+		waitUntilBastionsDeleted = g.Add(flow.Task{
+			Name:         "Waiting until Bastions have been deleted",
+			Fn:           flow.TaskFn(botanist.WaitUntilBastionsDeleted).SkipIf(shootSSHAccessEnabled),
+			Dependencies: flow.NewTaskIDs(deleteBastions),
+		})
```

Comment:
Why do we have to wait for it?

Reply:
Good point, this task is not needed. I will delete the botanist.WaitUntilBastionsDeleted function as well since I do not use it anywhere else.